pkg/kds/global/components.go

```diff
@@ -133,7 +147,11 @@ func Setup(rt runtime.Runtime) error {
 	onZoneToGlobalSyncConnect := mux.OnZoneToGlobalSyncConnectFunc(func(stream mesh_proto.KDSSyncService_ZoneToGlobalSyncServer, errChan chan error) {
 		zoneID, err := util.ClientIDFromIncomingCtx(stream.Context())
 		if err != nil {
-			errChan <- err
+			select {
+			case errChan <- err:
+			default:
```

Comment:
why are we sometimes dropping the error and sometimes logging it?

Reply:
I thought I was missing logger, I've added logging